test/integrations/hyperdrive/PriceDiscovery.t.sol

```diff
@@ -0,0 +1,673 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.20;
+
+import { AssetId } from "contracts/src/libraries/AssetId.sol";
+import { FixedPointMath, ONE } from "contracts/src/libraries/FixedPointMath.sol";
+import { HyperdriveMath } from "contracts/src/libraries/HyperdriveMath.sol";
+import { HyperdriveTest, HyperdriveUtils, IHyperdrive } from "test/utils/HyperdriveTest.sol";
+import { Lib } from "test/utils/Lib.sol";
+
+contract PriceDiscoveryTest is HyperdriveTest {
+    using FixedPointMath for uint256;
+    using HyperdriveUtils for *;
+    using Lib for *;
+
+    function test_solvency_at_0_apr(
+        uint256 fixedAPR,
+        uint256 initialContribution,
+        uint256 addLiquidityContribution
+    ) external {
+        uint256 minimumShareReserves = 10e18;
+
+        // Normalize the fuzzing parameters to a reasonable range.
+        fixedAPR = fixedAPR.normalizeToRange(0.01e18, 2e18);
```

Comment:
I'm wondering whether it'd be helpful to use the same fuzzing parameters across all tests... like all fuzz tests that modify fixedAPR use the same lower and upper bound? Initial thinking is that it'd increase confidence in pools with those ranges at the cost of decreased specificity.

Reply:
I like having different ranges for different tests. A lot of the tests (all of the `test_priceDiscovery` tests) are doing the same things with different parameters. By having both, we get to fuzz across the full range but we also concentrate.